packages/block-editor/src/components/list-view/index.js

```diff
@@ -50,15 +50,22 @@ export const BLOCK_LIST_ITEM_HEIGHT = 36;
 /**
  * Show a hierarchical list of blocks.
  *
- * @param {Object}  props                 Components props.
- * @param {string}  props.id              An HTML element id for the root element of ListView.
- * @param {Array}   props.blocks          Custom subset of block client IDs to be used instead of the default hierarchy.
- * @param {boolean} props.showBlockMovers Flag to enable block movers
- * @param {boolean} props.isExpanded      Flag to determine whether nested levels are expanded by default.
- * @param {Object}  ref                   Forwarded ref
+ * @param {Object}  props                   Components props.
+ * @param {string}  props.id                An HTML element id for the root element of ListView.
+ * @param {Array}   props.blocks            Custom subset of block client IDs to be used instead of the default hierarchy.
+ * @param {boolean} props.showBlockMovers   Flag to enable block movers
+ * @param {boolean} props.isExpanded        Flag to determine whether nested levels are expanded by default.
+ * @param {Object}  props.MoreMenuComponent Optional more menu substitution.
+ * @param {Object}  ref                     Forwarded ref
  */
 function ListView(
-	{ id, blocks, showBlockMovers = false, isExpanded = false },
+	{
+		id,
+		blocks,
+		showBlockMovers = false,
+		isExpanded = false,
+		MoreMenuComponent,
```

Comment:
An upper-case first letter for a prop doesn't seem right, I think the pattern is usually to keep the prop with a lower case letter and then alias it to seem more component-like:
```suggestion
		moreMenu: MoreMenuComponent,
```

Also, I notice it's called `MoreMenu` here, but it's called `BlockSettingsMenu` everywhere else in the codebase.

It'd be good to make this private as well.